src/components/textBox/2_clone/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { useRef, useEffect } from 'react';
+import cx from '../cx';
+import { clamp } from '@/components/utils';
+
+const MIN_LINES = 3;
+const MAX_LINES = 15;
+
+// TODO textbox, lineclamp 공통 사용 로직 hook으로 분리
+const TextBox = () => {
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const cloneRef = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    const elem = textareaRef.current;
+    const cloneElem = cloneRef.current;
+
+    const handleInput = () => {
+      if (!elem || !cloneElem) return;
+
+      const value = elem.value;
+      cloneElem.value = value;
+
+      const totalRows = Math.floor(
+        cloneElem.scrollHeight / cloneElem.clientHeight,
+      );
+      const clampedRows = clamp(totalRows, MIN_LINES, MAX_LINES);
+
+      elem.rows = clampedRows;
+    };
+
+    if (elem) elem.addEventListener('input', handleInput);
+
+    return () => {
+      if (elem) elem.removeEventListener('input', handleInput);
+    };
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    const elem = textareaRef.current;
    const cloneElem = cloneRef.current;
    if (!elem) {
        return undefined;
    }

    const handleInput = () => {
      if (!cloneElem) return;

      const value = elem.value;
      cloneElem.value = value;

      const totalRows = Math.floor(
        cloneElem.scrollHeight / cloneElem.clientHeight,
      );
      const clampedRows = clamp(totalRows, MIN_LINES, MAX_LINES);

      elem.rows = clampedRows;
    };

    elem.addEventListener('input', handleInput);

    return () => elem.removeEventListener('input', handleInput);
  }, []);
```
이렇게 최상단에서 미리 필터링 해버리면 뒤에서 안 해줘도 된답니다!